concordium-consensus/src/Concordium/ImportExport.hs

```diff
@@ -153,87 +288,202 @@ exportDatabaseV3 ::
+                                                            -- of the database.
+                                                            _ :|> (gh, _) -> gh
+                                                    if genHash /= exportedGenHash
+                                                        then do
+                                                            liftIO . putStrLn $
+                                                                "Genesis blockhash '"
+                                                                    <> show exportedGenHash
+                                                                    <> "' of most recently exported genesis index "
+                                                                    <> "does not match genesis blockhash '"
+                                                                    <> show genHash
+                                                                    <> "' found in node database for genesis index '"
+                                                                    <> show genIndex
+                                                                    <> "'"
+                                                            return (True, Empty)
+                                                        else do
+                                                            chunks <-
+                                                                writeChunks
+                                                                    genIndex
+                                                                    (demoteProtocolVersion (protocolVersion @pv))
+                                                                    genHash
+                                                                    startHeight
+                                                                    (_bpHeight . sbInfo $ sb)
+                                                                    outDir
+                                                                    chunkSize
+                                                                    lastWrittenChunkM
+                                                            return (False, singleton (genHash, chunks))
+                                        )
+                                        (DBState dbh 0)
                     closeDatabase dbh
-                    exportSections dbDir outDir indexHdl chunkSize (genIndex + 1) 1
-        else putStrLn ("The tree state database does not exist at " ++ treeStateDir)
+                    -- if an error occurred, return sections
+                    -- that were succesfully written to the
+                    -- file system; otherwise export the section
+                    -- corresponding to the incremented genesis
+                    -- index
+                    if exportError
+                        then return sectionData
```

Comment:
It would be good that the program terminates with non-zero status code if an error occurs.

That way they can more easily monitor it. So this error would ideally be propagated outwards to `main`.